### ISSUE TITLE
skip already downloaded files using download command

### ISSUE THREAD

**Issue by postyr** [prasmussen/gdrive#353]
HI 

When I use the command:
`gdrive download -r --path "path_to_local_folder" <googledrive folder id>`

It stops when it reaches a file I have already downloaded when running the command earlier.

When I use the command:
`gdrive download -r -f --path "path_to_local_folder" <googledrive folder id>`

It overwrites all the files and downloads everything again.

The Question:
Is it possible to skip files already downloaded, and how?

I have installed gdrive using Homebrew on Mac OS X 10.11.6.

**Comment by e-beach** [comment]
@postyr 
I don't think this is possible using `gdrive`. I'm in the process of forking this project at https://github.com/e-beach/gdrive, would you be interested if I implemented this feature as a flag?

**Comment by warrendodge** [comment]
I hit the same thing. A real pit after going for some time and it quits.
Rather than -f and the above have a way to add characters to any duplicate file so everything is downloaded and kept.

When taking pictures on 2 phones I have had the same file name uploaded from each phone. Then when I download the error occurs. The -f isn't right since the files are different pictures. Skipping isn't right either. Something like  file.jpg and then gdrive0_file.jpg gdrive1_file.jpg ...for duplicate files from the drive. Some messages as to what it is doing too so you know to look for it.

**Comment by SIAAAAAA** [comment]
I face the same thing. When I try to download a set of large files in a link, it raises `Failed to get file: googleapi: Error 403: Rate Limit Exceeded, rateLimitExceeded`. Then it seems that I need to re download everything. Is there anyone find a way to resolve it?

**Comment by Qwerty-Space** [comment]
@elliott-beach did you implement skipping existing local files?

**Comment by Qwerty-Space** [comment]
Oh

**Comment by warrendodge** [comment]
I want to chime in here since someone is looking at the code, I assume.

I have seen two things that make this gdrive unhappy

1. On a CIFS file system it seems to miss some of the file.
    I debugged as best I could and found that the GO call  that asks for 
a file list is the culprit. I could not debug deeper than that.
    Currently I copy the CIFS tree to /tmp and then use gdrive and it works.

2. It appears error or warning or retry conditions are not being 
processed properly.
     When I tried to do many file transactions it would not work properly.
      It failed at different places in the set of files.
      It seemed to have a problem with long paths at one time.
      I tried a number of things to no avail.
      I assume the GOOGLE API has changed or not implemented properly

     I instead wrote a script to get a file listing and process each 
file one at a time.
     I added all kinds of checks for problems and was able to get that 
to work.
     I am attaching a perl file with the way I made things work for 
large numbers of files.
     The download routines is where to look.

Here is what I am using
gdrive version
gdrive: 2.1.0
Golang: go1.6
OS/Arch: linux/amd64

**Comment by Qwerty-Space** [comment]
@warrendodge it doesn't appear that you've attached anything

**Comment by Qwerty-Space** [comment]
Also, there's a `--skip` flag in gdrive.go, but it's not implemented

**Comment by d-diaz** [comment]
Anybody found a workaround to this yet?

**Comment by amadeusrotundo** [comment]
> Anybody found a workaround to this yet?

compile from source, --skip works.

**Comment by byrsapty** [comment]
> > Anybody found a workaround to this yet?
> 
> compile from source, --skip works.

Tell me more please, otherwise when I upload a folder it creates copies of files (1) (2) every time and so on
